src/patch.js

```diff
@@ -22,6 +22,33 @@ var walker = require('./walker'),
     getWalker = walker.getWalker,
     setWalker = walker.setWalker;
 
+/**
+ * @const {boolean}
+ */
+var IS_PRODUCTION = process.env.NODE_ENV === 'production';
+
+if (!IS_PRODUCTION) {
+  var assertNoUnclosedTags = function(root) {
```

Comment:
The [announcement](https://medium.com/google-developers/introducing-incremental-dom-e98f79ce2c5f#5e85) had this as one of Incremental Dom's features compared to virtual DOM renderers:

> Both those assumptions are not true in a generic templating language: E.g. it may allow you to put some <section> tag in one template, then render a few children, but never close that <section> tag.